hierarchicalforecast/methods.py

```diff
@@ -398,27 +416,53 @@ def reconcile(self,
         elif self.method == 'wls_struct':
             W = np.diag(S @ np.ones((n_bottom,)))
         elif self.method in res_methods:
-            #we need residuals with shape (obs, n_hiers)
+            # Residuals with shape (obs, n_hiers)
             residuals = (y_insample - y_hat_insample).T
             n, _ = residuals.shape
+
+            # Protection: against overfitted model
+            residuals_sum = np.sum(residuals, axis=0)
+            zero_residual_prc = np.abs(residuals_sum) < 1e-4
+            zero_residual_prc = np.mean(zero_residual_prc)
+            if zero_residual_prc > .98:
+                raise Exception(f'Insample residuals close to 0, zero_residual_prc={zero_residual_prc}. Check `Y_df`')
+
+            # Protection: cases where data is unavailable/nan
             masked_res = np.ma.array(residuals, mask=np.isnan(residuals))
             covm = np.ma.cov(masked_res, rowvar=False, allow_masked=True).data
+
             if self.method == 'wls_var':
                 W = np.diag(np.diag(covm))
             elif self.method == 'mint_cov':
                 W = covm
             elif self.method == 'mint_shrink':
+                # Schäfer and Strimmer 2005, scale invariant shrinkage
+                # lasso or ridge might improve numerical stability but
+                # this version follows https://robjhyndman.com/papers/MinT.pdf
                 tar = np.diag(np.diag(covm))
+
+                # Protection: constant's correlation set to 0
                 corm = cov2corr(covm)
-                xs = np.divide(residuals, np.sqrt(np.diag(covm)))
+                corm = np.nan_to_num(corm, nan=0.0)
+
+                # Protection: standardized residuals 0 where residual_std=0
+                residual_std =  np.sqrt(np.diag(covm))
```

Comment:
If you want to preserve the `return_std` argument of `cov2corr`, please let's use its output here instead of computing `residual_std` twice. Something like,

```
corm, residual_std = cov2corr(cov, return_std=True)
```

Reply:
ibid.